src/base/torrentfileguard.h

```diff
@@ -50,6 +53,10 @@ class FileGuard
 class TorrentFileGuard
 {
     Q_GADGET
+    // moc from Qt4 ignores Q_ENUMS when it is behind #if QT_VERSION check
+    // this declaration is needed for Qt 4 only
+    // TODO Qt5: remove when dropping Qt4 support
+    Q_ENUMS(AutoDeleteMode)
```

Comment:
The qt5 code path calls both `Q_ENUMS()` and `Q_ENUM()`. Is that ok?
(I am taking your word for it).